components/script/dom/bindings/str.rs

```diff
@@ -306,13 +306,28 @@ impl DOMString {
     /// A valid number is the same as what rust considers to be valid,
     /// except for +1., NaN, and Infinity.
     /// https://html.spec.whatwg.org/multipage/#valid-floating-point-number
-    pub fn is_valid_number_string(&self) -> bool {
+    pub fn is_valid_float_number_string(&self) -> bool {
         let input = &self.0;
         input.parse::<f64>().ok().map_or(false, |val| {
             !(val.is_infinite() || val.is_nan() || input.ends_with(".") || input.starts_with("+"))
         })
     }
 
+    /// https://html.spec.whatwg.org/multipage/#best-representation-of-the-number-as-a-floating-point-number
+    pub fn best_representation_of_the_float_number(&mut self) {
+        let input = &self.0.clone();
+        match input.parse::<f64>() {
```

Comment:
Can we re-use the logic in `is_valid_float_number_string`, or refactor so the logic isn't duplicated?

Reply:
Sure, I would like to reuse if possible.
I did think about that, but I had no idea at that time.
If you have a good idea, please tell me. :)

Reply:
@NLincoln  I have re-factored it. :)